### PR TITLE
MDT CGM support

### PR DESCRIPTION
**Pull request by scottleibrand**
Initial attempt at MDT CGM support, based on commands provided by @bender1061.  John (or anyone else with Enlite), can you pull the oref0-setup branch and test whether this actually works?  I only have Dexcom myself, so I've just been able to test that the script itself works, not whether the resulting loop works.

### CODE REVIEW

**Comment by johnmales** [comment]
Hi Scott, Great progress but a few issues:
Openaps looping successfully, temp basals initiated appropriately. NS OpenAPS pill, Pump status pill updating.
Some NS visualization issues  - "Purple line" appears, but CGM values not uploading to nightscout. This may be a timezone issue at my end, I will look at it further.

carbhistory.json is empty apart from `[]`
Running `openaps ns-loop` gives the following output: 
```
openaps ns-loop
Starting ns-loop at Tue 20 Sep 22:10:17 AEST 2016:
ns://JSON/shell/cgm/ns-glucose.json
reporting cgm/ns-glucose.json
settings/autosens.json
Refreshed temptargetsWarning: could not parse monitor/carbhistory.json
Warning: Argument order has changed: please update your oref0-meal device and meal.json report to place carbhistory.json after basalprofile.json

/usr/local/lib/node_modules/oref0/lib/meal/history.js:19
    for (var i=0; i < carbHistory.length; i++) {
                                 ^
TypeError: Cannot read property 'length' of undefined
    at findMealInputs (/usr/local/lib/node_modules/oref0/lib/meal/history.js:19:34)
    at generate (/usr/local/lib/node_modules/oref0/lib/meal/index.js:8:20)
    at Object.<anonymous> (/usr/local/lib/node_modules/oref0/bin/oref0-meal.js:99:21)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:906:3
 and meal-carbs
Uploaded
```
I'm not sure if the warning about argument order is significant. Is the error in carbhistory due to it being empty?

I hope this helps I'll look at it further as well.

**Comment by dougtakeuti** [comment]
I can test it tomorrow night with the Pi + TI stick rig. Would that help?

Em 20/09/2016 01:08, "Scott Leibrand" <notifications@github.com> escreveu:

**Comment by scottleibrand** [comment]
@dougtakeuti Yes, it would, thanks. Please do share your results even if we end up merging it by then.

**Comment by scottleibrand** [comment]
@johnmales have you ever entered (or uploaded) carbs into NS?  If not, you might be hitting a bootstrap bug.  Try putting in a carb entry and see if that clears it up.

**Comment by Bender1061** [comment]
Looks like it worked good for me. at least for the stuff that I gave Scott. 

Yes, we do have nightscout visualization issues. I have been unable to get the CGM data to upload to nightscout. What is needed to make this work is not there. And I've been unable to figure out how to format the data correctly to make it work. 

OpenAPS pill does works well. One of the advantages of this setup, is you do not need to be hooked up to the internet in order for your loop you run. (really helps me, considering I sometimes work at plants that cell service is flaky, or sometimes, it's hard to keep the phone charged cause the battery drains fast while being a hotspot)  

Anyone been able to get cgm data sent to nightscout to work? Or know what needs to be added?

File: lib/iob/index.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ function generate (inputs) {
     };
 
     var iobArray = [];
+    console.error(inputs.clock);
+    if (! /(Z|[+-][0-2][0-9]:[03]0)+/.test(inputs.clock) ) {
```
Comment (elnjensen) [inline lib/iob/index.js]: I'd be inclined to make that regex just a little more general, maybe - possible there are some non-half-hour timezones?  Could just do something like \d\d:\d\d (though I don't know if JS uses that syntax for regex digits).  Most likely fine as-is, though.

Comment (scottleibrand) [inline lib/iob/index.js]: Heh, according to https://en.wikipedia.org/wiki/List_of_time_zone_abbreviations Nepal actually uses a :45 offset, so updated to allow that too without a warning.  :-)

Comment (elnjensen) [inline lib/iob/index.js]: Nice!  Yes, several 45-minute offsets:  https://www.timeanddate.com/time/time-zones-interesting.html

**Comment by scottleibrand** [comment]
Just updated to have get-bg run as part of pump-loop, not as a standalone cgm-loop that might fight over the radio stick with the pump-loop.  Can a few people please re-test to confirm that works right?

**Comment by dougtakeuti** [comment]
It would be good to remind older users to update mmeowlink, as this might be an issue with the wait-for-silence alias.

**Comment by scottleibrand** [comment]
@dougtakeuti would be better to upgrade automatically if needed in https://github.com/openaps/oref0/blob/oref0-setup/bin/oref0-setup.sh#L188-L193

**Comment by scottleibrand** [comment]
Per @johnmales  https://github.com/openaps/oref0/pull/221#issuecomment-249448374 - "So far working with MDT cgm setup - temp basals being set appropriately, openaps pill in NS reporting correctly."

**Pull request merged**